File: SocketComm.Examples/SendValue/IntClient.cs
using static SocketComm.Cmd;

namespace SocketComm.Examples.SendValue;

public class IntClient
{
    public static void Main()
    {
        FPipe p = new FPipe("/tmp/hello.pipe");
        p.Connect();
        p.WriteCmd(Ready);
        Cmd c;
        c = p.ReadCmd();
        while (c == Receive)
        {
            int i = p.ReceiveValue(SimpleConversion.BytesToNumber<int>);
            Console.WriteLine($"Received {i}");
            p.WriteCmd(Ready);
            p.ExpectCmd(Go);
            i = Bench(i);
            p.WriteCmd(Done);
            Console.WriteLine($"Sending {i}");
            p.SendValue(i, SimpleConversion.NumberToBytes);
            c = p.ReadCmd();
        }
        p.Dispose();
    }

    public static int Bench(int i)
    {
        return i * 2;
    }
}

File: SocketComm.Examples/SendValue/IntClient.java
import java.nio.ByteBuffer;
public class IntClient {
    public static void main(String[] args) throws Exception {
        FPipe p = new FPipe("/tmp/intvalue.pipe");
        p.WriteCmd(Cmd.Ready);
        Cmd c;
        c = p.ReadCmd();
        while (c == Cmd.Receive)
        {
            int i = p.ReceiveValue(ByteBuffer::getInt);
            System.out.printf("Received %d\n", i);
            p.WriteCmd(Cmd.Ready);
            p.ExpectCmd(Cmd.Go);
            i = Bench(i);
            p.WriteCmd(Cmd.Done);
            System.out.printf("Sending %d\n", i);
            p.SendValue(i, (val)-> ByteBuffer.allocate(4).putInt(val));
            c = p.ReadCmd();
        }
        p.close();
    }

    public static int Bench(int i){
        return i * 2;
    }
}
